gravitee-gateway-services/gravitee-gateway-services-heartbeat/src/main/java/io/gravitee/gateway/services/heartbeat/HeartbeatService.java

```diff
@@ -130,20 +130,27 @@ protected void doStart() throws Exception {
             super.doStart();
             LOGGER.info("Start gateway heartbeat");
 
+            vertx.eventBus().registerCodec(new RepositoryManagementEventCodec());
             heartbeatEvent = prepareEvent();
             topic.publish(heartbeatEvent);
 
-            // Remove the state to not include it in the underlying repository as it's just used for internal
-            // purpose
-            heartbeatEvent.getProperties().remove(EVENT_STATE_PROPERTY);
```

Comment:
This line was the reason for those errors that we saw when running the code with another thread